Test/tutorial/Simple_compiler/Compiler.dfy

```diff
@@ -0,0 +1,671 @@
+// RUN: %dafny /compile:0 /spillTargetCode:3 "%s"
+// RUN: dotnet run --project %S/SimpleCompiler.csproj -- %S/example_input.calc > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+/// A simple compiler pipeline
+/// ==========================
+///
+/// This file implements and verifies the Dafny part of the compiler pipeline
+/// documented in README.md.
+///
+/// Source language
+/// ---------------
+
+module DafnyAST {
+
+/// We have two kinds of AST nodes:
+///
+/// 1. Expressions:
+///    Constants, variable accesses, or binary operations.
+
+  datatype BinOp =
+    | Add
+    | Sub
+
+  datatype Expr =
+    | Const(n: int)
+    | Var(v: string)
+    | Op(op: BinOp, e1: Expr, e2: Expr)
+
+/// 2. Statements:
+///     Either no-ops or print statements (or a sequence of two statements)
+
+  datatype Stmt =
+    | Skip
+    | Print(e: Expr)
+    | Assign(v: string, e: Expr)
+    | Seq(s1: Stmt, s2: Stmt)
+
+/// The semantics of expressions and statements can be defined using recursive interpreters:
+
+  type Context = map<string, int>
+
+  function method interpExpr(e: Expr, ctx: Context): int {
```

Comment:
Is `interpret` shortened to `interp` so it's not longer than `compile` ?

Reply:
It's a pretty common abbreviation, but I'm not sure why it's used ^^  Probably length, indeed?